Yoast/javascript
# Ignore prefixes for Hebrew keyphrase recognition

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Creates basic keyphrase forms for Hebrew when they're preceded by one of the following prefixes: `ב ה ו כ ל מ ש`.
* [Yoast SEO Free] Makes it possible to recognize keyphrases in Hebrew when they're preceded by the most common articles or prepositions (for example: `עולם` in `העולם`).

## Relevant technical choices:

* The current implementation is based on creating all possible prefixed forms.
* `getWordForms` is always called. The place where morphology/non-morphology support is decided on is within `getWordForms`.  Specifically, `determineStem` will return the input word instead of a stem if there is no stem function or no morphology data available. The rest of `getWordForms` is also run when there is no morphology support. That's why it's possible to add the basic word forms within the `getWordForms` function.
* Usually written without nikud (vowel markers) and also other markers (e.g. distinguishing between shin and sin), see https://www.memrise.com/course/702355/hebrew-reading-and-language-intro/7/. Therefore, only the base forms of the letters were included in the list of prefixes.

## Test instructions

This PR can be tested by following these steps:
Test the following functionality in Free:
* Set your site to Hebrew.
* As your keyphrase, set the word `ראשית` ("beginning").
* Add a text with at least 150 words.
* In the text, add the form `בראשית` ("in the beginning").
* The keyphrase density assessment should show 1 keyphrase occurrence. When clicking on the marker for the assessmet, `בראשית` should be marked.
* Switch around the keyphrase and the word form.
* The keyphrase density assessment should still show 1 keyphrase occurrence.
* Switch the form in the text to a form with another prefix, e.g. `לראשית` ("to the beginning").
* The keyphrase density assessment should still show 1 keyphrase occurrence.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LIN-417
